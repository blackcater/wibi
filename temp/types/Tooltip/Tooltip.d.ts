import { VPopper } from '../popper';
import { RenderX } from '../utils';

export declare class Tooltip extends VPopper {
  /**
   * content of tooltip
   *
   * @default: ''
   **/
  title: RenderX;

  /**
   * inline mode of tooltip
   *
   * @default: false
   **/
  inline: boolean;

  /**
   * white theme of tooltip
   *
   * @default: false
   **/
  white: boolean;

  /**
   * inner width of tooltip
   *
   * @default: 250
   **/
  maxWidth: number;

  /**
   * inner style of tooltip
   *
   * @default: {}
   **/
  innerStyle: object;

  /**
   * Value of positionLeft
   *
   * @default: 0
   **/
  positionLeft?: number;

  /**
   * Value of positionTop
   *
   * @default: 2
   **/
  positionTop?: number;
}
