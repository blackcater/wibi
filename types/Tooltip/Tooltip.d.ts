import { VPopper } from '../popper';

export declare class Tooltip extends VPopper {
  /**
   * content of tooltip
   *
   * @default: ''
   **/
  title: string;

  /**
   * inline mode of tooltip
   *
   * @default: false
   **/
  inline: boolean;

  /**
   * white theme of tooltip
   *
   * @default: false
   **/
  white: boolean;

  /**
   * inner width of tooltip
   *
   * @default: 250
   **/
  maxWidth: number;

  /**
   * inner style of tooltip
   *
   * @default: {}
   **/
  innerStyle: object;

  /**
   * Value of positionLeft
   *
   * @default: 0
   **/
  positionLeft?: number;

  /**
   * Value of positionTop
   *
   * @default: 2
   **/
  positionTop?: number;
}
