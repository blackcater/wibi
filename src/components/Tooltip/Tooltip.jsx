import VueTypes from 'vue-types';
import popperMixin from 'mixins/popper';
import prefix, { defaultClassPrefix } from 'utils/prefix';
import { addStyle, getAttr } from 'shares/dom';

const CLASS_PREFIX = 'tooltip';

export default {
  name: 'Tooltip',

  model: {
    prop: 'visible',
    event: 'change',
  },

  mixins: [popperMixin],

  props: {
    title: VueTypes.string,
    theme: VueTypes.oneOf(['dark', 'light']).def('dark'),
    maxWidth: VueTypes.number.def(250),
    classPrefix: VueTypes.string.def(defaultClassPrefix(CLASS_PREFIX)),
  },

  computed: {
    popperClasses() {
      return [
        this._addPrefix('popper'),
        {
          [this._addPrefix(`popper-theme-${this.theme}`)]: this.theme,
        },
      ];
    },
    innerClasses() {
      return [
        this._addPrefix('inner'),
        {
          [this._addPrefix('inner-with-width')]: !!this.maxWidth,
        },
      ];
    },
  },

  // FIXME: check dynamic change the title property
  watch: {
    currentVisible(val) {
      const popper = this.$refs.popper;
      const placement = getAttr(popper, 'x-placement');

      if (!val && popper) {
        if (~placement.indexOf('top')) {
          addStyle(popper, { top: '2px' });
        }

        if (~placement.indexOf('bottom')) {
          addStyle(popper, { top: '-2px' });
        }

        if (~placement.indexOf('right')) {
          addStyle(popper, { left: '-2px' });
        }

        if (~placement.indexOf('left')) {
          addStyle(popper, { left: '2px' });
        }
      }
    },
  },

  render() {
    const tooltipData = {
      class: this.classPrefix,
      directives: [{ name: 'click-outside', value: this._handleClickOutside }],
      on: {},
      ref: 'container',
    };
    const referenceData = {
      class: this._addPrefix('rel'),
      on: {},
      ref: 'reference',
    };
    const popperData = {
      class: this.popperClasses,
      style: {},
      directives: [
        { name: 'show', value: this.currentVisible },
        { name: 'transfer-dom' },
      ],
      attrs: {
        'data-transfer': `${this.transfer}`,
      },
      ref: 'popper',
    };
    const arrowData = {
      class: this._addPrefix('arrow'),
      ref: 'arrow',
    };

    this._addTriggerListeners(referenceData, tooltipData);

    return (
      <div {...tooltipData}>
        <div {...referenceData}>{this.$slots.default}</div>
        <transition name="tooltip-fade">
          <div {...popperData}>
            <div {...arrowData} />
            <div class={this.innerClasses}>{this.title}</div>
          </div>
        </transition>
      </div>
    );
  },

  methods: {
    _addPrefix(cls) {
      return prefix(this.classPrefix, cls);
    },
  },
};
