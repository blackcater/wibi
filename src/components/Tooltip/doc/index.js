export default ['basic', 'placement', 'trigger'];
