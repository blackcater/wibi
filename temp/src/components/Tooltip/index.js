import Tooltip from './Tooltip.jsx';

Tooltip.install = function(Vue) {
  Vue.component(Tooltip.name, Tooltip);
};

export default Tooltip;
