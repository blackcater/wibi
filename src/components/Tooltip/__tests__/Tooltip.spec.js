import { mount } from '@vue/test-utils';
import Tooltip from 'components/Tooltip';

describe('Tooltip component', () => {
  it('should render correctly', () => {
    const wrapper = mount({
      render() {
        return <Tooltip inline title="demo text" />;
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
