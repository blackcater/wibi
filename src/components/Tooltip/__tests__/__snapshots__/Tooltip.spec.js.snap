// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip component should render correctly 1`] = `
<div x-placement="auto" class="vs-tooltip-popper vs-tooltip-popper-inline">
  <div class="vs-tooltip-arrow"></div>
  <div class="vs-tooltip-inner" style="max-width: 250px;">demo text</div>
</div>
`;
